go/database/mpt/proof_test.go

```diff
@@ -175,6 +175,101 @@ func TestCreateWitnessProof_CanCreateProof(t *testing.T) {
 	}
 }
 
+func TestCreateWitnessProof_ProofContainsEmptyNodes(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	address := common.Address{1}
+	key := common.Key{2}
+
+	ctxt := newNodeContextWithConfig(t, ctrl, S5LiveConfig)
+	addressNibbles := AddressToNibblePath(address, ctxt)
+	keyNibbles := KeyToNibblePath(key, ctxt)
+
+	tests := map[string]struct {
+		desc NodeDesc
+	}{
+		"account storage empty": {
+			desc: &Branch{
+				children: Children{
+					addressNibbles[0]: &Branch{
+						children: Children{
+							addressNibbles[1]: &Extension{
+								path: addressNibbles[2:50],
+								next: &Account{address: address, pathLength: 14, info: AccountInfo{common.Nonce{1}, amount.New(1), common.Hash{0xAA}},
+									storage: &Empty{}}},
+						}}},
+			},
+		},
+		"path incomplete - branch": {
+			desc: &Branch{
+				children: Children{
+					addressNibbles[0]: &Empty{},
+					addressNibbles[0] - 1: &Account{address: address, pathLength: 14, info: AccountInfo{common.Nonce{1}, amount.New(1), common.Hash{0xAA}},
+						storage: &Empty{}},
+				},
+			},
+		},
+		"path incomplete - extension": {
+			desc: &Branch{
+				children: Children{
+					addressNibbles[0]: &Extension{}},
+			},
+		},
```

Comment:
Both of these are invalid MPTs -- there must not be a Child node with only one successor, and every extension must have a non-empty next.